src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Cmdlet/UpdateAzureSqlManagedInstance.cs

```diff
@@ -0,0 +1,190 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.Sql.Common;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Security;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    /// <summary>
+    /// Defines the Update-AzureRmSqlManagedInstance cmdlet
+    /// </summary>
+    [Cmdlet(VerbsData.Update, "AzureRmSqlManagedInstance",
+        SupportsShouldProcess = true,
+        ConfirmImpact = ConfirmImpact.Medium)]
+    public class UpdateAzureSqlManagedInstance : ManagedInstanceCmdletBase
+    {
+        protected const string UpdateByNameAndResourceGroupParameterSet =
+            "Update a Managed Instance from cmdlet input parameters";
+
+        protected const string UpdateByInputObjectParameterSet =
+            "Update a Managed Instance from AzureSqlManagedInstanceModel instance definition";
+
+        protected const string UpdateByResourceIdParameterSet =
+            "Update a Managed Instance from an Azure resource id";
+
+        /// <summary>
+        /// Gets or sets the name of the Managed instance to use.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = UpdateByNameAndResourceGroupParameterSet, 
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "SQL Database Managed instance name.")]
+        [Alias("Name")]
+        [ValidateNotNullOrEmpty]
+        public string ManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = UpdateByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// The new SQL administrator password for the Managed instance.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The new SQL administrator password for the Managed instance.")]
+        [ValidateNotNull]
+        public SecureString AdministratorPassword { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Managed instance License Type
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Determines which License Type of Sql Azure Managed Instance to use")]
+        [ValidateSet(Constants.LicenseTypeBasePrice, Constants.LicenseTypeLicenseIncluded, IgnoreCase = false)]
```

Comment:
Use PSArgumentValidator to allow user to type in custom values for forward compatibility

Reply:
Same comment (PSArgumentCompleter)

Reply:
Done